### PR TITLE
feat: populate default settings, warn user if key missing

### LINKED ISSUE #16: populate default settings, warn user about file location if key missing
thought of a more fleshy config. Nevertheless, i like the key(s) to be set as a rule of thumb, waiting for values to be filled.
(the user is blindfolded, otherwise sees that there are keys without a (proper) value)

_Originally posted by @emphasize in https://github.com/OpenVoiceOS/skill-ovos-fallback-chatgpt/issues/15#issuecomment-1890475071_

### PR DESCRIPTION
**Pull request by mikejgray**
closes #16

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -75,13 +86,15 @@ def _async_ask(self, message):
             for utt in self.chat.stream_utterances(utterance):
                 answered = True
                 self.speak(utt)
-        except: # speak error on any network issue / no credits etc
-            pass
+        except Exception as err: # speak error on any network issue / no credits etc
+            self.log.error(err)
+            self.speak_dialog("gpt_error")
```
Comment (JarbasAl) [inline __init__.py]: duplicated dialog

File: __init__.py [hunk 2]
```diff
@@ -18,14 +23,20 @@ def runtime_requirements(self):
         )
 
     def initialize(self):
+        self.settings.merge(DEFAULT_SETTINGS, new_only=True)
+        self.unconfigured_message = f"ChatGPT not configured yet, please set your API key in {self.settings_path}"
```
Comment (JarbasAl) [inline __init__.py]: any specific reason this is a variable? we dont usually do that for logs so curious what you had in mind

Comment (mikejgray) [reply __init__.py]: The biggest reason is that it's logged in two places, so it made sense to make it a variable. Especially since it's so long

Comment (JarbasAl) [reply __init__.py]: I only see it once in in line 96 ?

Comment (mikejgray) [reply __init__.py]: Ah right, I just deleted one 🙂

**Review by JarbasAl — APPROVED** [review]

**Pull request merged**